shell/platform/embedder/embedder_task_runner.cc

```diff
@@ -18,9 +18,12 @@ EmbedderTaskRunner::EmbedderTaskRunner(DispatchTable table,
           fml::MessageLoopTaskQueues::GetInstance()->CreateTaskQueue()) {
   FML_DCHECK(dispatch_table_.post_task_callback);
   FML_DCHECK(dispatch_table_.runs_task_on_current_thread_callback);
+  FML_DCHECK(dispatch_table_.destruction_callback);
 }
 
-EmbedderTaskRunner::~EmbedderTaskRunner() = default;
+EmbedderTaskRunner::~EmbedderTaskRunner() {
```

Comment:
When is this called? Is it something explicit like Deinitialize or Shutdown, so that we aren't in a situation where we have an unbroken retain loop because `FlutterEngine` transitively owns `EmbedderTaskRunner` and causes it to live as long as the engine itself?

Reply:
Addressed in offline discussion; it is indeed triggered in `Shutdown`, which means it's happening via in the same high-level call as the current cleanup.